app/templating/metadata_context.py

```diff
@@ -4,58 +4,37 @@
 
 def build_metadata_context(metadata):
     """
-    :param metadata: user metadata
+    :param metadata: token metadata
     :return: metadata context
     """
-    render_data = {
-        'survey': _build_survey_meta(metadata),
-        'respondent': _build_respondent_meta(metadata),
-    }
-    return render_data
-
-
-def _build_respondent_meta(metadata):
-    respondent_id = None
-    name = None
-    trading_as = None
-
-    if metadata:
-        respondent_id = metadata['ru_ref']
-        name = metadata['ru_name']
-        trading_as = metadata['trad_as']
-
-    respondent_meta = {
-        'tx_id': convert_tx_id(metadata['tx_id']),
-        'respondent_id': json_and_html_safe(respondent_id),
-        'name': json_and_html_safe(name),
-        'trading_as': json_and_html_safe(trading_as),
-    }
-    return respondent_meta
-
-
-def _build_survey_meta(metadata):
     return {
-        'return_by': metadata['return_by'],
-        'start_date': metadata['ref_p_start_date'],
-        'end_date': metadata['ref_p_end_date'],
-        'employment_date': metadata['employment_date'],
-        'region_code': json_and_html_safe(metadata['region_code']) if 'region_code' in metadata else None,
-        'period_str': json_and_html_safe(metadata['period_str']),
         'eq_id': json_and_html_safe(metadata['eq_id']),
-        'collection_id': json_and_html_safe(metadata['collection_exercise_sid']),
         'form_type': json_and_html_safe(metadata['form_type']),
+        'ref_p_start_date': metadata.get('ref_p_start_date'),
```

Comment:
Is there a reason that we're still hard-coding these keys into runner? Shouldn't we be able to iterate through the keys in the survey `required_metadata` and put those values into the context?

Reply:
To be honest, I just wasn't quite sure the purpose of this card was to also decouple runner from these metadata. I thought it was to ensure we have what the required metadata to launch and submit the survey successfully.

But that is definitely a good idea and I'll make the changes soon. Since `required_metadata` only contains the metadata for the schema and not things like `eq_id`, shall I handle this the same way I did for `parse_metadata`?
Also, would it be inefficient to just make use of the `parse_metadata` method again to return all the key/values instead of iterating through again?